packages/vite/src/node/build.ts

```diff
@@ -508,11 +514,14 @@ async function doBuild(
     }
 
     if (Array.isArray(outputs)) {
-      const res = []
-      for (const output of outputs) {
-        res.push(await generate(output))
-      }
-      return res
+      // the first output is for legacy bundle which should be excuted first
+      return [
+        await generate(outputs[0]),
+        ...(await usePromiseAll<OutputOptions, RollupOutput>(
```

Comment:
**thought:** Even if the first may be for legacy, isn't it better maybe to have a flag for _needs sequential_ + order/prio?
Something like `'pre'` and `'post'` we already have for plugins

The current solution doesn't satisfy me so much :confused:

Reply:
>They work the same way and have the same types as Build Hooks **but are called separately for each call to bundle.generate(outputOptions) or bundle.write(outputOptions)**  ----- https://rollupjs.org/guide/en/#output-generation-hooks

 in output-generate part, you just pass through the different format output you want then you get the corresponding bundle, so they are isolated by each other, so we can do it concurrently here

but in vite, we intentionly put the legacy output at first 
https://github.com/vitejs/vite/blob/df46c0d9af709a03e6964a7f33565039bc1aba3e/packages/plugin-legacy/index.js#L157

and we need to get polyfill injected in every entry chunk, so we need to generate `legacy bundle` first, in short, it's a `vite-case` here, because they shared variables through clousure
https://github.com/vitejs/vite/blob/df46c0d9af709a03e6964a7f33565039bc1aba3e/packages/plugin-legacy/index.js#L267


**References**:
-https://rollupjs.org/guide/en/#how-do-i-add-polyfills-to-a-rollup-bundle